src/editor/Collaboration.js

```diff
@@ -125,6 +137,27 @@ define(function (require, exports, module) {
             }
         }
         _changing = false;
+        peer.on('fileTransfer', function (metadata, receiver) {
+            console.log('incoming filetransfer', metadata.name, metadata);
+            receiver.on('progress', function (bytesReceived) {
+                console.log("Percentage of file received : " + (bytesReceived / metadata.size * 100));
+            });
+            receiver.on('receivedFile', function (file, metadata) {
+                var reader = new window.FileReader();
+                reader.readAsArrayBuffer(file);
+                reader.onloadend = function() {
```

Comment:
I think this will work (untested):

```js
reader.onload = function(e) {
    var data = e.target.result;
    var buffer = new FilerUtils.Buffer(data); // you forgot to use `new` below.
    var filename = Path.join(StartupState.project("root"), metadata.name);
    _received[metadata.name] = true;
    FilerUtils.writeFileAsBinary(filename, buffer, function(err) {
        if(err) {
            console.log(err);
        };
        delete _received[metadata.name];
    });
};
// Move this below the event handler defn
reader.readAsArrayBuffer(file);
```

Also, you need to handle the case of the reader having an error.